eli5/formatters/as_dataframe.py

```diff
@@ -0,0 +1,102 @@
+from itertools import chain
+from singledispatch import singledispatch
+from typing import Dict, List, Optional
+import warnings
+
+import pandas as pd  # type: ignore
+
+from eli5.base import (
+    Explanation, FeatureImportances, TargetExplanation, TransitionFeatureWeights,
+)
+
+
+EXPORTED_ATTRIBUTES = ['transition_features', 'targets', 'feature_importances']
+
+
+def format_as_dataframes(expl):
+    # type: (Explanation) -> Dict[str, pd.DataFrame]
+    """ Export an explanation to a dictionary with ``pandas.DataFrame`` values
+    and string keys that correspond to explanation attributes.
+    Use this method if several dataframes can be exported from a single explanation
+    (e.g. for CRF explanation with has both feature weights and transition matrix).
+    """
+    result = {}
+    for attr in EXPORTED_ATTRIBUTES:
+        value = getattr(expl, attr)
+        if value:
+            result[attr] = format_as_dataframe(value)
+    return result
+
+
+@singledispatch
+def format_as_dataframe(expl):
+    # type: (Explanation) -> Optional[pd.DataFrame]
+    """ Export an explanation to a single ``pandas.DataFrame``.
+    In case several dataframes could be exported by
+    :func:`eli5.formatters.as_dataframe.format_as_dataframes`,
+    a warning is raised. If no dataframe can be exported, ``None`` is returned.
+    This function also accepts some components of the explanation as arguments:
+    feature importances, targets, transition features.
+    """
+    for attr in EXPORTED_ATTRIBUTES:
+        value = getattr(expl, attr)
+        if value:
+            other_attrs = [a for a in EXPORTED_ATTRIBUTES
+                           if getattr(expl, a) and a != attr]
+            if other_attrs:
+                warnings.warn('Exporting {} to DataFrame, but also {} could be '
+                              'exported. Consider using eli5.format_as_dataframes.'
+                              .format(attr, ', '.join(other_attrs)))
+            return format_as_dataframe(value)
+
+
+@format_as_dataframe.register(FeatureImportances)
+def feature_importances_to_df(feature_importances):
```

Comment:
What do you think about making such functions private, as they are implementation details of format_.. functions?